### PR TITLE
Fixes issue #407: Fixes Reflections.getConstructor() method logic to detect constructors with parameters annotated with @Id or @Column

### PR DESCRIPTION
**Pull request by dearrudam**
Fixes: https://github.com/eclipse/jnosql/issues/407

## Changes

- Fixes Reflections.getConstructor() method logic to detect constructors with parameters annotated with @Id or @Column

### CODE REVIEW

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

File: jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java [hunk 1]
```diff
@@ -24,14 +24,12 @@
...
+import java.lang.annotation.Annotation;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.Modifier;
 import java.lang.reflect.Parameter;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
+import java.util.*;
```
Comment (otaviojava) [inline jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java:32]: Please, avoid wild imports; keep them one by one.

File: jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java [hunk 2]
```diff
@@ -148,10 +146,17 @@ void makeAccessible(Field field) {
      * @throws ConstructorException when the constructor has public and default
      */
     public <T> Constructor<T> getConstructor(Class<T> type) {
+
+        final Predicate<String> isIdAnnotation = Id.class.getName()::equals;
```
Comment (otaviojava) [inline jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java:150]: How about moving those to static?

File: jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java [hunk 3]
```diff
@@ -148,10 +146,17 @@ void makeAccessible(Field field) {
...
+        final Predicate<String> isColumnAnnotation = Column.class.getName()::equals;
+
         final Predicate<Constructor<?>> defaultConstructorPredicate = c -> c.getParameterCount() == 0;
         final Predicate<Constructor<?>> customConstructorPredicate = c -> {
             for (Parameter parameter : c.getParameters()) {
-                if (parameter.getAnnotation(Id.class) != null || parameter.getAnnotation(Column.class) != null) {
+                if (Arrays.stream(parameter.getAnnotations())
```
Comment (otaviojava) [inline jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java:156]: This condition increased, how about refactoring to a method that says explicitly what is going on here, such as "hasNoSQLAnnotation"

**Comment by otaviojava** [comment]
@dearrudam did you find one way to reproduce it with unit tests?

File: jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java (hunk 2 above)
Comment (dearrudam) [reply jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java:150]: Yeah... making these predicates static ones will avoid creating Predicate instances ever this method is executed...

**Comment by dearrudam** [comment]
> @dearrudam did you find one way to reproduce it with unit tests?

No, @otaviojava. I don't know how to reproduce it on our side... as this issue was detected in Quarkus integration by now, my suspicion is that it's related to something that Quarkus is doing under the hood maybe... but, I think that this PR fixes this issue without problems according to the test suite implemented on the project (in special this [test](https://github.com/dearrudam/jnosql/blob/2318ac7e902c6a5f89930a437cbc04e46454591d/jnosql-mapping/jnosql-mapping-core/src/test/java/org/eclipse/jnosql/mapping/reflection/ReflectionsTest.java#L66)) and by considering the evidence that I described on this [comment](https://github.com/eclipse/jnosql/issues/407#issuecomment-1627832904) also.

**Comment by otaviojava** [comment]
Merged, thank you @dearrudam

**Pull request merged**